docs/app_modules/interchain-accounts/integration.md

```diff
@@ -0,0 +1,129 @@
+// If a submodule is being statically disabled, the associated Keeper does not need to be added. 
+type App struct {
+    ...
+
+    ICAControllerKeeper icacontrollerkeeper.Keeper
+	ICAHostKeeper       icahostkeeper.Keeper
+    ICAAuthKeeper       icaauthkeeper.Keeper
+
+    ...
+}
+
+...
+
+// Create store keys for each submodule Keeper and the authentication module
+keys := sdk.NewKVStoreKeys(
+    ...
+    icacontrollertypes.StoreKey,
+	icahosttypes.StoreKey,
+    icaauthtypes.StoreKey,
+    ...
+)
+
+... 
+
+// Create the scoped keepers for each submodule keeper and authentication keeper
+scopedICAControllerKeeper := app.CapabilityKeeper.ScopeToModule(icacontrollertypes.SubModuleName)
+scopedICAHostKeeper := app.CapabilityKeeper.ScopeToModule(icahosttypes.SubModuleName)
+scopedICAAuthKeeper := app.CapabilityKeeper.ScopeToModule(icaauthtypes.ModuleName)
+
+...
+
+// Create the Keeper for each submodule
+app.ICAControllerKeeper = icacontrollerkeeper.NewKeeper(
+		appCodec, keys[icacontrollertypes.StoreKey], app.GetSubspace(icacontrollertypes.SubModuleName),
+		app.IBCKeeper.ChannelKeeper, // may be replaced with middleware such as ics29 fee
+		app.IBCKeeper.ChannelKeeper, &app.IBCKeeper.PortKeeper,
+		app.AccountKeeper, scopedICAControllerKeeper, app.MsgServiceRouter(),
+)
+app.ICAHostKeeper = icahostkeeper.NewKeeper(
+		appCodec, keys[icahosttypes.StoreKey], app.GetSubspace(icahosttypes.SubModuleName),
+		app.IBCKeeper.ChannelKeeper, &app.IBCKeeper.PortKeeper,
+		app.AccountKeeper, scopedICAHostKeeper, app.MsgServiceRouter(),
+)
+
+// Create Interchain Accounts AppModule
+icaModule := ica.NewAppModule(&app.ICAControllerKeeper, &app.ICAHostKeeper)
+
+// Create your Interchain Accounts authentication module
+app.ICAAuthKeeper = icaauthkeeper.NewKeeper(appCodec, keys[icaauthtypes.StoreKey], app.ICAControllerKeeper, scopedICAAuthKeeper)
+
+// ICA auth AppModule
+icaAuthModule := icaauth.NewAppModule(appCodec, app.ICAAuthKeeper)
+
+// ICA auth IBC Module
+ICAAuthIBCModule := icaauth.NewIBCModule(app.ICAAuthKeeper)
+
+// Create host and controller IBC Modules as desired
+icaControllerIBCModule := icacontroller.NewIBCModule(app.ICAControllerKeeper, icaAuthIBCModule)
+icaHostIBCModule := icahost.NewIBCModule(app.ICAHostKeeper)
+
+// Register host and authentication routes
+ibcRouter.AddRoute(icacontrollertypes.SubModuleName, icaControllerIBCModule).
+		AddRoute(icahosttypes.SubModuleName, icaHostIBCModule).
+		AddRoute(icaauthtypes.ModuleName, icaControllerIBCModule) // Note, the authentication module is routed to the top level of the middleware stack
+
+...
+
+// Register Interchain Accounts and authentication module AppModule's
+app.moduleManager = module.NewManager(
+    ...
+    icaModule,
+    icaAuthModule,
+)
+
+...
+
+// Add Interchain Accounts module InitGenesis logic
+app.mm.SetOrderInitGenesis(
+    ...
+    icatypes.ModuleName,
+    ...
+)
+```
+
+### Interchain Accounts on-chain parameters
+
+#### Example Initialization of parameters
```

Comment:
I added some params docs [here](https://hackmd.io/9Fj34HF4R-GNBwuM2VI3Vg?both) let me know if you'd like me to add them here after this is merged.

Reply:
yup! Lets add after this is merged